src/common/io.ts

```diff
@@ -113,10 +118,79 @@ export class ARIO {
     }
     return new ARIOReadable(config);
   }
+
+  static mainnet(): AoARIORead;
+  static mainnet(
+    config: ARIOConfigNoSigner & { faucetUrl?: string },
+  ): AoARIORead;
+  static mainnet(
+    config: ARIOConfigWithSigner & { faucetUrl?: string },
+  ): AoARIORead;
```

Comment:
This should return a write interface if the config is the WithSigner variant

Reply:
[4f942ae](https://github.com/ar-io/ar-io-sdk/pull/451/commits/4f942aeee4f69a426e81e2cd9503a9c5e3bad645)

Reply:
good catch!